legend-sdlc-generation-service/src/main/java/org/finos/legend/sdlc/generation/service/ServiceExecutionClassGenerator.java

```diff
@@ -67,50 +68,56 @@ private ServiceExecutionClassGenerator(Service service, String packagePrefix, St
     public GeneratedJavaClass generate()
     {
         String packageName = generatePackageName();
-        List<ExecutionParameter> executionParameters = getExecutionParameters();
-        MutableMap<String, Object> dataModel = Maps.mutable
-                .with(
-                        "classPackage", packageName,
-                        "service", this.service,
-                        "planResourceName", this.planResourceName,
-                        "streamProviderParameterName", getStreamProviderParameterName(executionParameters)
-                ).withKeyValue("imports", getImports(executionParameters))
+        MutableList<ExecutionParameter> executionParameters = getExecutionParameters();
+        MutableMap<String, Object> dataModel = Maps.mutable.<String, Object>empty()
+                .withKeyValue("classPackage", packageName)
+                .withKeyValue("service", this.service)
+                .withKeyValue("planResourceName", this.planResourceName)
+                .withKeyValue("streamProviderParameterName", getStreamProviderParameterName(executionParameters))
+                .withKeyValue("imports", getImports(executionParameters))
                 .withKeyValue("executionParameters", executionParameters);
 
-        String freeMarkerTemplate = "generation/service/ServiceExecutionClassGenerator.ftl";
-        InputStream inputStream = Objects.requireNonNull(
-                Thread.currentThread().getContextClassLoader().getResourceAsStream(freeMarkerTemplate),
-                () -> "Unable to find freemarker template '" + freeMarkerTemplate + "' on context classpath"
-        );
-        try (InputStreamReader reader = new InputStreamReader(inputStream))
+        DefaultObjectWrapper objectWrapper = new DefaultObjectWrapper(Configuration.VERSION_2_3_30);
+        objectWrapper.setExposeFields(true);
+        objectWrapper.setExposureLevel(BeansWrapper.EXPOSE_ALL);
+        Template template = loadTemplate();
+        try
         {
-            DefaultObjectWrapper objectWrapper = new DefaultObjectWrapper(Configuration.VERSION_2_3_30);
-            objectWrapper.setExposeFields(true);
-            objectWrapper.setExposureLevel(BeansWrapper.EXPOSE_ALL);
-            Template template = new Template("ServiceExecutionClassGenerator", reader, new Configuration(Configuration.VERSION_2_3_30));
             StringWriter code = new StringWriter();
             template.process(dataModel, code, objectWrapper);
-            return new GeneratedJavaClass(packageName + "." + this.service.name, code.toString());
+            // Use \n for all line breaks to ensure consistent behavior across environments
+            String codeString = code.toString().replaceAll("\\R", "\n");
```

Comment:
will this cause CRLF `\r\n` to become 2 new lines?

Reply:
No, the pattern `\R` should understand `\r\n` as a single line break. So it should be replaced with a single `\n.`